dask/array/linalg.py

```diff
@@ -328,3 +329,98 @@ def svd(a, name=None):
     dask.array.linalg.tsqr: Actual implementation with citation
     """
     return tsqr(a, name, compute_svd=True)
+
+
+def lu(a):
+    import scipy.linalg
+
+    if a.ndim != 2:
+        raise ValueError('dimension must be 2 to perform lu decomposition')
+
+    xdim, ydim = a.shape
+    if xdim != ydim:
+        raise ValueError('input must be a square matrix to perform lu decomposition')
+    if not len(set(a.chunks[0] + a.chunks[1])) == 1:
+        raise ValueError('all chunks must be a square matrix to perform lu decomposition')
```

Comment:
`a.chunks` ideally has the following form: `((n, n, n, n, n), (n, n, n, n, n))` for some integer `n`.  This check looks sane to me given that form.